Behaviors/FrameworkElementBehaviors.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Collections.Generic;
+using System.Windows;
+using System.Windows.Data;
+using Thingie.WPF.Converters;
+
+namespace Thingie.WPF.Behaviors
+{
+    /// <summary>
+    /// Represents behaviors that can be applied to a <see cref="FrameworkElement"/>.
+    /// </summary>
+    public class FrameworkElementBehaviors
+    {
+        /// <summary>
+        /// A <see cref="DependencyProperty"/> used to store the element's <see cref="FrameworkElement"/> children. This provides a safe and controlled way to dynamically set the element's properties based on the registered children's properties.
+        /// </summary>
+        public static readonly DependencyProperty RegisteredChildrenProperty = 
+            DependencyProperty.RegisterAttached("RegisteredChildren", typeof(List<FrameworkElement>), typeof(FrameworkElementBehaviors), new PropertyMetadata(null, OnRegisteredChildrenChanged));
+
+        /// <summary>
+        /// Adds the specified <see cref="FrameworkElement"/> value to the element's <see cref="RegisteredChildrenProperty"/>.
+        /// </summary>
+        /// <param name="element">The element to whose <see cref="RegisteredChildrenProperty"/> the <see cref="FrameworkElement"/> value will be added.</param>
+        /// <param name="value">The <see cref="FrameworkElement"/> to be added to the <see cref="RegisteredChildrenProperty"/>.</param>
+        public static void SetRegisteredChildren(FrameworkElement element, FrameworkElement value)
+        {
```

Comment:
This is quite confusing.  Are we setting children or adding a child?

Reply:
Also goofed.